### PR TITLE
implementation started

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/routes/patient.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  getOneField,
+  getPatientById,
+  getTotalBalance,
+  getTotalDues,
+  updatePatientDetails,
+} from "controllers/patient";
+
+const patientRouter = express.Router();
+
+patientRouter.route("/:_id/:field").get(getOneField);
```
Comment (rehanbaber1710) [inline src/routes/patient.ts]: add asyncHandler here

File: src/controllers/patient.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  validateObjectId,
+  validatePatient,
+} from "helper";
+import asyncHandler from "middleware/asyncHandler";
+import Appointment from "models/appointment";
+import Patient from "models/patient";
+
+export const getOneField = asyncHandler(async (req: Request, res: Response) => {
+  const { _id, field } = req.params;
+  if (!(await validateObjectId(_id))) throwException(res, "Invalid Id", 400);
```
Comment (rehanbaber1710) [inline src/controllers/patient.ts]: Create a function to check Id validation and throw error from that function

File: src/controllers/patient.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  const result: any = await Patient.findById(_id);
+  if (!result) throwException(res, "Invalid Id", 400);
+  if (!result[field]) throwException(res, "Invalid field", 400);
```
Comment (rehanbaber1710) [inline src/controllers/patient.ts]: move if condition from this code and create a function to check null or empty array.

File: src/controllers/patient.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  sendAndLog(
+    res,
+    `${field} for\nId: ${_id} and Name: ${result.name} is:\n${result[field]}`
+  );
+});
+
+export const addPatient = asyncHandler(async (req: Request, res: Response) => {
```
Comment (rehanbaber1710) [inline src/controllers/patient.ts]: move aysncHandler to router level

File: src/startup/db.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { Express } from "express";
```
Comment (rehanbaber1710) [inline src/startup/db.ts]: move this to models folder

File: src/startup/logger.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import winston from "winston";
```
Comment (rehanbaber1710) [inline src/startup/logger.ts]: move this to helper folder

File: src/startup/routes.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import express, { Express } from "express";
```
Comment (rehanbaber1710) [inline src/startup/routes.ts]: move the logic to main index.ts file

File: src/data/index.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (rehanbaber1710) [inline src/data/index.ts:1]: what is this? 
1- It should be the part of helper.
2- Why we only have 1 error code & message?

File: src/models/appointment.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import mongoose from "mongoose";
+
+const appointmentSchema = new mongoose.Schema({
+  sTime: { type: Number, required: true, min: 0, max: 23 },
```
Comment (rehanbaber1710) [inline src/models/appointment.ts:4]: what is sTime?

File: src/models/appointment.ts [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  eTime: {
```
Comment (rehanbaber1710) [inline src/models/appointment.ts:5]: what is eTime?

File: src/models/appointment.ts [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    type: Number,
+    required: true,
+    min: 1,
+    max: 24,
+  },
+  desc: {
```
Comment (rehanbaber1710) [inline src/models/appointment.ts:11]: what is desc?

File: src/models/appointment.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    trim: true,
+    minLength: 3,
+    maxLength: 35,
+  },
+  fee: {
+    type: Number,
+    default: 0,
+    min: 0,
+  },
+  pId: {
```
Comment (rehanbaber1710) [inline src/models/appointment.ts:23]: what is pId? I can't understand the variable names

File: src/startup/routes.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
+// import express, { Express } from "express";
```
Comment (rehanbaber1710) [inline src/startup/routes.ts:1]: remove this folder

File: src/data/index.ts (hunk 8 above)
Comment (ghost) [reply src/data/index.ts:1]: this is the default message. Otherwise, the message and both codes are provided when the error is thrown.

**Pull request merged**